source/module_hamilt_lcao/hamilt_lcaodft/LCAO_matrix.cpp

```diff
@@ -877,6 +878,13 @@ void LCAO_Matrix::destroy_HS_R_sparse(void)
         HR_sparse[0].swap(empty_HR_sparse_up);
         HR_sparse[1].swap(empty_HR_sparse_down);
         SR_sparse.swap(empty_SR_sparse);
+        if(GlobalV::ESOLVER_TYPE == "tddft" && elecstate::H_TDDFT_pw::stype ==1)
```

Comment:
we will delete LCAO_matrix.cpp in near future, so please don't add any new codes in LCAO_matri.cpp, find some where or even construct a new class/namespace

Reply:
I see. I have no idea what the new solution without LCAO_Matrix would be. But whether it count in the special case of tddft HR or not, this fix will definitely be conflict with it and will have to be adjusted. This fix was meant to be a tempoary one, so I thought a fix in a consisent form with the original code would be easier to be considered in during the refactoring work. But nevermind, I'll set up a new class to store the data.

Reply:
Great!